packages/playwright-core/src/server/webkit/wkBrowser.ts

```diff
@@ -357,4 +358,26 @@ export class WKBrowserContext extends BrowserContext {
     if (process.platform === 'win32' && this._browser.options.headful && (viewportSize.width < 250 || viewportSize.height < 240))
       throw new Error(`WebKit on Windows has a minimal viewport of 250x240.`);
   }
+
+  private async _polyfillPublicKeyCredential() {
+    if (process.platform === 'darwin' || process.platform === 'win32')
+      return;
+
+    function polyfill() {
+      // https://developer.mozilla.org/en-US/docs/Web/API/PublicKeyCredential
+      globalThis.PublicKeyCredential ??= {
+        async getClientCapabilities() {
+          return {};
+        },
+        async isConditionalMediationAvailable() {
+          return false;
+        },
+        async isUserVerifyingPlatformAuthenticatorAvailable() {
+          return false;
+        },
+      } as any;
+    }
+
+    await this.addInitScript(`(${polyfill})()`);
```

Comment:
Instead of using the public API that has some side-effects, do it similar to other things in `_calculateBootstrapScript()`.